### PR TITLE
don't log to file by default

### PR DESCRIPTION
**Pull request by wochinge**
**Proposed changes**:
- API requests are not longer logged to ``rasa_core.log`` by default in order to avoid
  problems when running on OpenShift

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -21,6 +21,8 @@ Changed
 -------
 - new event broker class: ``SQLProducer``. This event broker is now used when running locally with
   Rasa X
+- API requests are not longer logged to ``rasa_core.log`` by default in order to avoid
+  problems when running on OpenShift
```
Comment (erohmensing) [inline CHANGELOG.rst]: Maybe with a little migration here -- "to get logs to this file pass the `--log_file rasa_core.log` parameter or some other way to let people know that this setting has changed? A log upon server start or something? Just worried about people who rely on this default and then don't realize when it stops logging

Comment (wochinge) [reply CHANGELOG.rst]: added it to the changelog, not worth adding it to the migration guide imo

**Comment by wochinge** [comment]
> If we want to cut a release for them, do we have to cut rasa and then rasa x bc of compatibility?

`rasa` would be enough, but due to the changes here https://github.com/RasaHQ/rasa/pull/4097 we should also release a `rasa-x` including https://github.com/RasaHQ/rasa-x/pull/1034

**Comment by erohmensing** [comment]
sounds good

**Pull request merged**